backend/sublet/views.py

```diff
@@ -62,11 +71,40 @@ class Properties(viewsets.ModelViewSet):
     """
 
     permission_classes = [SubletOwnerPermission | IsSuperUser]
-    serializer_class = SubletSerializer
+    parser_classes = (MultiPartParser, FormParser, JSONParser)
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
+            return SubletSerializerRead
+        return SubletSerializer
 
     def get_queryset(self):
         return Sublet.objects.all()
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)  # Check if the data is valid
+        instance = serializer.save()  # Create the Sublet
+        instance_serializer = SubletSerializerRead(instance=instance, context={"request": request})
+        return Response(instance_serializer.data, status=status.HTTP_201_CREATED)
+
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop("partial", False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+
+        queryset = self.filter_queryset(self.get_queryset())
+        # no clue what this does but I copied it from the DRF source code
+        if queryset._prefetch_related_lookups:
```

Comment:
lol uh probably look into this since we shouldn't deploy code we don't understand (idek fully what this is doing)

Reply:
will do, but that method is just a copy paste of the DRF source doe with a minor change to the return